controllers/main.go

```diff
@@ -1119,73 +1130,88 @@ func (controller *MainController) AdminTicketsPost(c web.C, r *http.Request) (st
+	}
+
+	action := strings.ToLower(r.PostFormValue("action"))
+	switch action {
+	case "add", "remove":
+		// recognized action
 	default:
 		session.AddFlash("invalid or unknown form action type", "adminTicketsError")
 		return "/admintickets", http.StatusSeeOther
 	}
 
-	actionVerb := "Unknown"
-	switch r.PostFormValue("action") {
-	case "Add":
+	actionVerb := "unknown"
+	switch action {
+	case "add":
 		actionVerb = "added"
 		ignoredLowFeeTickets, err := controller.StakepooldGetIgnoredLowFeeTickets()
 		if err != nil {
-			session.AddFlash("GetIgnoredLowFeeTickets error: "+err.Error(), "adminTicketsError")
+			session.AddFlash("GetIgnoredLowFeeTickets error: "+err.Error(),
+				"adminTicketsError")
 			return "/admintickets", http.StatusSeeOther
 		}
 
-		for _, ticketToAddString := range r.PostForm["tickets[]"] {
-			t := time.Now()
+		for i, tickethash := range ticketHashes {
+			t := ticketList[i]
 
 			// TODO check if it is already present in the database
 			// and error out if so
 
-			tickethash, err := chainhash.NewHashFromStr(ticketToAddString)
-			if err != nil {
-				session.AddFlash("NewHashFromStr failed for "+ticketToAddString+": "+err.Error(), "adminTicketsError")
-				return "/admintickets", http.StatusSeeOther
-			}
-
-			msa, exists := ignoredLowFeeTickets[*tickethash]
+			msa, exists := ignoredLowFeeTickets[tickethash]
 			if !exists {
-				session.AddFlash("ticket " + ticketToAddString + " is no longer present")
+				session.AddFlash("ticket " + t + " is no longer present")
 				return "/admintickets", http.StatusSeeOther
 			}
 
 			expires := controller.CalcEstimatedTicketExpiry()
 
 			lowFeeTicket := &models.LowFeeTicket{
-				AddedByUid:    session.Values["UserId"].(int64),
+				AddedByUid:    userID,
 				TicketAddress: msa,
-				TicketHash:    ticketToAddString,
+				TicketHash:    t,
 				TicketExpiry:  0,
 				Voted:         0,
-				Created:       t.Unix(),
+				Created:       time.Now().Unix(),
 				Expires:       expires.Unix(),
 			}
 
-			if err := models.InsertLowFeeTicket(dbMap, lowFeeTicket); err != nil {
-				session.AddFlash("Database error occurred while adding ticket "+ticketToAddString, "adminTicketsError")
+			err = models.InsertLowFeeTicket(dbMap, lowFeeTicket)
+			if err != nil {
+				session.AddFlash("Database error occurred while adding ticket "+
+					t, "adminTicketsError")
 				log.Warnf("Adding ticket %v failed: %v", tickethash, err)
 				return "/admintickets", http.StatusSeeOther
 			}
 		}
-	case "Remove":
+
+	case "remove":
 		actionVerb = "removed"
-		query := "DELETE FROM LowFeeTicket WHERE TicketHash IN (\"" +
-			strings.Join(r.PostForm["tickets[]"], ",") + "\")"
-		_, err := dbMap.Exec(query)
+		// To use gorm's slice expansion, use a mapper with a ticketList. For
+		// three tickets in the list, gorm will expand this to:
+		//     "... IN (:Tickets0,:Tickets1,:Tickets2)".
+		// This allows each string in the list two be it's own argument.
```

Comment:
```suggestion
		// This allows each string in the list to be it's own argument.
```

Reply:
Good catch.  Also "it's" is not right here.